### DISCUSSION TITLE
How to use multi BackgroundTasks

### DISCUSSION THREAD

**Discussion by batman-do** [fastapi/fastapi#10489] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

I want like this but not work
![image](https://github.com/tiangolo/fastapi/assets/77484083/6ec8fb07-8f48-4d34-8876-17f2c36bb236)

How to use like that? Thanks everyone

**Comment by n8sty** [comment, marked as answer]
What exactly are you trying to accomplish by doing this? Without checking, my assumption is that even if this were possible both `background_tasks_1` and `background_tasks_2` would refer to the same object.

**Reply by batman-do** [reply]
@n8sty  I have 2 independent tasks and want them to run concurrent in a post method like the generate_content function, I have to run BackgroundTasks so that the service does not time out and it will run in the background while I have another api that retrieves its results through a global variable. but I tried 2 TH
- Case 1: As shown in the picture, it doesn't work (can't create 2 backgrounds
- Case 2: using only 1 background for both of my tasks, it doesn't run at the same time, but runs one after the other, whichever is added first is executed first like that and I think it bug of fastapi

![image](https://github.com/tiangolo/fastapi/assets/77484083/7249998e-3a87-4dce-9146-a224d7500f2c)
